android/samples/sample-gltf-viewer/src/main/java/com/google/android/filament/gltf/MainActivity.kt

```diff
@@ -105,6 +122,73 @@ class MainActivity : Activity() {
         return ByteBuffer.wrap(bytes)
     }
 
+    private fun setStatusText(text: String) {
+        runOnUiThread {
+            Toast.makeText(applicationContext, text, Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    private suspend fun loadGlb(message: RemoteServer.IncomingMessage) {
+        withContext(Dispatchers.Main) {
+            modelViewer.destroyModel()
+            modelViewer.loadModelGlb(message.buffer)
+            modelViewer.transformToUnitCube()
+        }
+    }
+
+    private suspend fun loadZip(message: RemoteServer.IncomingMessage) {
+        val zipfileBytes = ByteArray(message.buffer.remaining())
+        message.buffer.get(zipfileBytes)
+
+        var gltfPath: String? = null
+        val pathToBufferMapping = withContext(Dispatchers.IO) {
+            val deflater = ZipInputStream(ByteArrayInputStream(zipfileBytes))
+            val mapping = HashMap<String, Buffer>()
+            while (true) {
+                val entry = deflater.nextEntry ?: break
+                if (entry.isDirectory) continue
+                if (entry.name.startsWith("__MACOSX")) continue
```

Comment:
Is `__MACOSX` not recognized as a directory? I wonder if `.DS_Store` files would cause a similar issue.

Reply:
This is really just an optimization, maybe it's questionable, I noticed these junky things appearing when I create ZIP files with macOS, and there's no reason to percolate them all the way down into gltfio.  Good point, I also see `.DS_Store` files in there so I think I should ignore those too.